pkg/getter/getter.go

```diff
@@ -85,6 +86,10 @@ func WithTLSClientConfig(certFile, keyFile, caFile string) Option {
 type Getter interface {
 	// Get file content by url string
 	Get(url string, options ...Option) (*bytes.Buffer, error)
+	// Retrieve the filename given a source URL
+	Filename(url *url.URL, version string) string
```

Comment:
`Get` takes a string as an argument: `href`. Why do we need to discern the difference between the URL and the version here? Shouldn't this just accept an `href` as well?

Reply:
This is for the downloaded tgz to maintain the same naming as currently. Even if your image tag is `charts/nginx:commithash`, it will download to `nginx-{chartVersion}.tgz`. See the comment at https://github.com/helm/helm/pull/7613/files#diff-9069fe52469166bbcbe989af4b5ae3f5R85. The core problem is that we can't determine the filename to download to from the URL alone, as the URL may not contain the chart version. Both the `URL` and `Filename` functions were added to avoid having to do a bunch of case-based code in the downloader (don't want the downloader to care if it's using a registry getter or some other getter).

Happy to consider alternatives here as well.